### PR TITLE
Add more unit tests for tools/processes.py

### PR DESCRIPTION
**Pull request by r0x0d**
The tool had a good coverage, but there were some tests missing, this patch solves this.

## Summary by CodeRabbit

* **Tests**
  * Reworked tests to be platform-agnostic using fixture-based mocks; expanded coverage for process listing and per-PID info, including current/missing PIDs, invalid inputs, float PID handling, and field truncation.
  * Added parametrized scenarios for remote host interactions, varied command outputs, error paths, and formatting edge cases. No production API changes; test-only.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tests/tools/test_processes.py [hunk 1]
```diff
@@ -2,56 +2,547 @@
...
+        mock_proc = mocker.Mock()
+        mock_proc.info = {"pid": 1, "name": "test"}
+
+        def process_iter_side_effect(*args, **kwargs):
+            # Return one good process and one that raises AccessDenied
+            yield mock_proc
+            mock_bad = mocker.Mock()
+            mock_bad.info
+            type(mock_bad).info = mocker.PropertyMock(side_effect=psutil.AccessDenied())
+            yield mock_bad
```
Comment (samdoran) [inline tests/tools/test_processes.py]: The coverage report indicates this isn't being called. I don't know why exactly but that's something that needs to be investigated.

It may be better to have separate functions, one that returns ok and another that returns AccessDenied and call them explicitly.

I'm also not sure about line 53. What is the point of `mock_bad.info`?

I'm also not really sure about line 54. Modifying the global state of the class seems to be asking for trouble even though this is the way the docs say to use `PropertyMock`.

It may be simpler to create a class with the desired behavior instead of using a Mock object.

```python
class Denier:
    def __init__(self, deny=False):
        self.deny = deny

    @property
    def info(self):
        if self.deny:
            raise psutil.AccessDenied

        return ["something", "useful"]
```

Then pass a list of those objects to `side_effect` and `return_value` isn't needed. The first time `pcocess_iter` is called a `Denier` object is returned that won't raise `AccessDenied` on `info` but the second and later calls to `process_into` will return `Denier` objects that raise when `info` is accessed.

```python
mocker.patch("psutil.process_iter", side_effect=[Denier(), Denier(True)]

```

Separate objects could be created instead of using a toggle, too.

File: tests/tools/test_processes.py [hunk 2]
```diff
@@ -2,56 +2,547 @@
...
+    async def test_list_processes_exception_handling(self, mocker):
+        """Test that list_processes handles general exceptions."""
+        mocker.patch("psutil.process_iter", side_effect=Exception("Test error"))
+
+        result = await mcp.call_tool("list_processes", {})
+        text = _get_text_from_mcp_result(result)
+        assert "Error listing processes" in text
+        assert "Test error" in text
 
     async def test_get_process_info_with_current_process(self):
```
Comment (samdoran) [inline tests/tools/test_processes.py:49]: It seems like many of these tests can be combined into a single test with different parameters.

File: tests/tools/test_processes.py [hunk 3]
```diff
@@ -2,56 +2,547 @@
...
+        """Test that get_process_info accepts float PIDs (from LLMs)."""
+        current_pid = float(os.getpid())
+        result = await mcp.call_tool("get_process_info", {"pid": current_pid})
+        text = _get_text_from_mcp_result(result)
+        assert isinstance(text, str)
+        assert str(int(current_pid)) in text
+
+    async def test_get_process_info_access_denied(self, mocker):
+        """Test get_process_info when access is denied."""
+        mock_proc = mocker.Mock(spec=psutil.Process)
```
Comment (samdoran) [inline tests/tools/test_processes.py]: Could this be a fixture instead of having to build it in the test?

File: tests/tools/test_processes.py [hunk 4]
```diff
@@ -2,56 +2,547 @@
...
+        text = _get_text_from_mcp_result(result)
+        assert "Error: PID must be at least 1" in text
+
+    async def test_list_processes(self):
+        """Test calling list_processes through mcp.call_tool()."""
+        result = await mcp.call_tool("list_processes", {})
+
+        # mcp.call_tool returns a tuple of (list[TextContent], dict)
+        assert isinstance(result, tuple)
+        assert len(result) == 2
```
Comment (samdoran) [inline tests/tools/test_processes.py]: I'm not sure these statements add much value. It would be better to do this once rather than in every test. The interface in FastMCP is totally different, which means all of these tests need to be rewritten. Reducing the number of times that needs to be done would be ideal.

File: tests/tools/test_processes.py [hunk 5]
```diff
@@ -2,56 +2,547 @@
...
+        # mcp.call_tool returns a tuple of (list[TextContent], dict)
+        assert isinstance(result, tuple)
+        content_list, metadata = result
+        assert isinstance(content_list, list)
+        content = content_list[0]
+        assert "Error: PID must be at least 1" in content.text
+
+    async def test_get_process_info_nonexistent_pid(self):
+        """Test calling get_process_info via MCP with nonexistent PID."""
+        result = await mcp.call_tool("get_process_info", {"pid": 999999})
```
Comment (samdoran) [inline tests/tools/test_processes.py]: Is there any way the `get_process_info` tests could be combined into one test with different parameters?

File: tests/tools/test_processes.py [hunk 6]
```diff
@@ -1,156 +1,253 @@
...
-Pid:	1
-PPid:	0
-Threads:	1
-VmRSS:	    11892 kB"""
-
-        mock_execute.side_effect = [
-            (0, ps_output, ""),  # ps command
-            (0, proc_status, ""),  # /proc/PID/status
+        assert "Running Processes" in result
+        assert "Total processes: 102" in result
```
[the remark this answers is not in the document]
Comment (major) [reply tests/tools/test_processes.py]: @r0x0d This is good stuff but you don't need to cover that in this commit.

File: tests/tools/test_processes.py [hunk 7]
```diff
@@ -1,156 +1,253 @@
...
 
-    async def test_list_processes_handles_command_failure(self, mock_execute):
+    async def test_list_processes_handles_command_failure(self, mock_execute_command):
         """Test that list_processes handles command failure gracefully."""
-        mock_execute.return_value = (1, "", "Command not found")
+        mock_execute_command.return_value = (1, "", "Command not found")
 
         result = await processes.list_processes(host="remote.host")
 
         assert "Error" in result
```
[the remark this answers is not in the document]
Comment (major) [reply tests/tools/test_processes.py]: This one looks easy to fix and deleting code feels nice.

[the remark this answers is not in the document]
Comment (major) [reply tests/tools/test_processes.py]: Uh I had no idea coderabbit was watching
